src/utils/event-manager/event-manager.js

```diff
@@ -0,0 +1,97 @@
+import {Manager} from 'hammerjs';
+import WheelInput from './wheel-input';
+import {
+  BASIC_EVENT_ALIASES, EVENT_RECOGNIZER_MAP, RECOGNIZERS, GESTURE_EVENT_ALIASES
+} from './constants';
+
+/**
+ * Single API for subscribing to events about both
+ * basic input events (e.g. 'mousemove', 'touchstart', 'wheel')
+ * and gestural input (e.g. 'click', 'tap', 'panstart').
+ * Delegates event registration and handling to Hammer.js.
+ */
+export default class EventManager {
+  constructor(element, options = {}) {
+    // TODO: support overriding default RECOGNIZERS by passing
+    // recognizers / configs, keyed to event name.
+
+    this._onBasicInput = this._onBasicInput.bind(this);
+    this.manager = new Manager(element, {recognizers: RECOGNIZERS})
+    .on('hammer.input', this._onBasicInput);
+
+    // Handle mouse wheel events as well
+    this.wheelInput = new WheelInput(element, event => {
+      this.manager.emit('wheel', event);
+    });
+  }
+
+  destroy() {
+    this.wheelInput.destroy();
+    this.manager.destroy();
+  }
+
+  /**
+   * Register an event handler function to be called on `event`.
+   * @param {string|Object} event   An event name (String) or map of event names to handlers
+   * @param {Function} [handler]    The function to be called on `event`.
+   */
+  on(event, handler) {
+    if (typeof event === 'string') {
+      // Special handling for gestural events.
+      const recognizerName = EVENT_RECOGNIZER_MAP[event];
+      if (recognizerName) {
+        // Enable recognizer for this event.
+        this.manager.get(recognizerName).set({enable: true});
+      }
+
+      // Alias to a recognized gesture as necessary.
+      const gestureEventAlias = GESTURE_EVENT_ALIASES[event];
+      // Register event handler.
+      this.manager.on(gestureEventAlias || event, handler);
```

Comment:
I see. This piece had a bug of firing the same event multiple times. I'll bring the wrapper back and try to resolve it.